Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -736,9 +740,32 @@ public static function buildPhpCommand(array $settings)
             $command .= ' -c ' . escapeshellarg($useIniFile);
         }
 
+        self::ensureCurrentPhpVersionMatchesConfiguredVersion($command);
+
         return $command;
     }
 
+    /**
+     * @param $command
+     * @throws Exception
+     */
+    protected static function ensureCurrentPhpVersionMatchesConfiguredVersion($command)
+    {
+        $runningPHPVersion = phpversion() . '-' . PHP_SAPI;
+        exec($command . ' -r "echo phpversion() . \'-\' . PHP_SAPI;"', $output, $result);
+
+        if ($result === 0) {
```

Comment:
an early return would improve readability IMO:
```
if ($result !== 0) {
    return;
}
```

Reply:
But then again - should we ignore that error silently?

Reply:
We check that elsewhere in the script:
https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Core/Booting/Scripts.php#L360-L375

so I didn't think it's necessary to duplicate that logic there.